### PR TITLE
Make sure that logits are always on the device where the model weights are

### PR DESCRIPTION
**Pull request by ManilShrestha**
The issue was due to the logits always being in device=0, I made it such that the logits generated by the model should always be in the model's device.

### CODE REVIEW

**Comment by rlouf** [comment]
What happens when the model is loaded on several GPUs? I think that was the rationale behind pinning the logits to device 0.

**Comment by rlouf** [comment]
Not that this is going to become irrelevant once #926 is merged.
